pkg/infrastructure/azure/azure.go

```diff
@@ -0,0 +1,356 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/msi/armmsi"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/coreos/stream-metadata-go/arch"
+	"github.com/sirupsen/logrus"
+	"k8s.io/utils/ptr"
+
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	"github.com/openshift/installer/pkg/rhcos"
+	aztypes "github.com/openshift/installer/pkg/types/azure"
+)
+
+// const (
+// 	tfVarsFileName         = "terraform.tfvars.json"
+// 	tfPlatformVarsFileName = "terraform.platform.auto.tfvars.json"
+// 	clusterOutputFileName  = "cluster.azuresdk.vars.json"
+
+// 	defaultDescription = "Created by Openshift Installer"
+// 	ownedTagKey        = "kubernetes.io/cluster/%s"
+// 	ownedTagValue      = "owned"
+// )
+
+// InfraProvider is the Azure SDK infra provider.
+type InfraProvider struct{}
+
+// InitializeProvider initializes the Azure SDK provider.
+func InitializeProvider() clusterapi.Provider {
+	return &InfraProvider{}
+}
+
+// Name returns the name of the provider.
+func (p *InfraProvider) Name() string {
+	return "azure"
+}
+
+// PreProvision is called before provisioning using CAPI controllers has begun.
+func (p *InfraProvider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error {
+	session, err := in.InstallConfig.Azure.Session()
+	if err != nil {
+		return fmt.Errorf("failed to get session: %w", err)
+	}
+	// sessionClient := icazure.NewClient(session)
+
+	installConfig := in.InstallConfig.Config
+	platform := installConfig.Platform.Azure
+	subscriptionID := session.Credentials.SubscriptionID
+
+	var cloudConfiguration cloud.Configuration
+	switch platform.CloudName {
+	case aztypes.PublicCloud:
+		cloudConfiguration = cloud.AzurePublic
+	case aztypes.USGovernmentCloud:
+		cloudConfiguration = cloud.AzureGovernment
+	}
```

Comment:
The `session` contains the cloud configuration so you should be able to use `session.CloudConfig`.

Reply:
Fixed.